### DISCUSSION TITLE
Package dependencies fail on scoped registry due to non-development dependencies

### DISCUSSION THREAD

**Discussion by joey-rieg-bl** [MixedRealityToolkit/MixedRealityToolkit-Unity#522] [General]
Hey :)

I was uploading the MRTK packages to our Verdaccio server for CI/CD and easier access. Unfortunately, the dependencies of the packages to other MRTK packages only stated the "release" version, e.g. `3.0.0`. But the all MRTK package are  named `x.x.x-development`. In the end I got some errors that the dependent package is not found.

Exact example:
`org.mixedrealitytoolkit.extendedassets 3.0.0-development` depends on `org.mixedrealitytoolkit.standardassets 3.0.0`:

```
"dependencies": {
    "org.mixedrealitytoolkit.standardassets": "3.0.0"
  }
```

`org.mixedrealitytoolkit.standardassets` package.json states the version to be `3.0.0-development`:

```
{
  "name": "org.mixedrealitytoolkit.standardassets",
  "version": "3.0.0-development",
  ...
}
```

I assume that UPM interprets `3.0.0-development` to be a lower version than `3.0.0` because when I changed the dependency to `3.0.0-development` the issue was resolved.

Do you handle those handle those renamings in your build pipelines? Is there a purpose of having a development version without a numbering system like Unity's `-pre.X` or `-exp.X` packages ?

**Comment by ghazen-ml** [comment]
Hi @joey-rieg-bl , the packages as represented in the github repo are currently a bit inconsistent since, as you say, they are noting their version to be 3.0.0-development, yet they depend on non-development versions. This could probably be fixed but would need some release work to accomplished.

There are a couple of workarounds you may want to try though for now:

1. Upload the official npm packages that were published instead of building new ones from the source tree. You can find these here: https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/releases. These are all versioned without "-development"
2. Edit the source tree before you generate npm packages to include the "-development" bit in the dependencies. This would allow unity to find the correct versions in your scoped registry.

**Reply by joey-rieg-bl** [reply]
Hey @ghazen-ml, thanks for the reply!

Okay, this makes a whole lot more sense now 😄. I was too focused on the source tree and totally missed the releases page. We already went for option 2. so we can automate the process and just add the "-development" via a Powershell script.

Thank you!
